feature_server/commands.py

```diff
@@ -979,42 +1014,30 @@ def handle_command(connection, command, parameters):
         command_func = commands[command]
     except KeyError:
         return # 'Invalid command'
+    aspec = command_func.argspec
+    min_params = len(aspec.args) - 1 - len(aspec.defaults or ())
+    max_params = len(aspec.args) - 1 if aspec.varargs is None else None
+    len_params = len(parameters)
+    if (len_params < min_params
+            or max_params is not None and len_params > max_params):
+        return 'Invalid number of arguments for %s' % command
     try:
-        if (hasattr(command_func, 'user_types') and
-            command_func.func_name not in connection.rights):
+        if not has_rights(command_func, connection):
                 return "You can't use this command"
         return command_func(connection, *parameters)
     except KeyError:
         return # 'Invalid command'
-    except TypeError:
-        return 'Invalid number of arguments for %s' % command
+    except TypeError as t:
+        print 'Command', command, 'failed with args:', parameters
+        print t
+        return 'Command failed'
     except InvalidPlayer:
         return 'No such player'
     except InvalidTeam:
         return 'Invalid team specifier'
     except ValueError:
         return 'Invalid parameters'
 
-def debug_handle_command(connection, command, parameters):
```

Comment:
The only place it seems to have been used is below it, and that's commented out. My guess it was once used a long time ago, but now it is well and truly redundant. (the `handle_command` function does way more than this function did anyway)